be/src/exec/olap_scan_node.cpp

```diff
@@ -186,9 +185,9 @@ Status OlapScanNode::get_next(RuntimeState* state, RowBatch* row_batch, bool* eo
 
     // check if Canceled.
     if (state->is_cancelled()) {
-        boost::unique_lock<boost::mutex> l(_row_batches_lock);
+        std::unique_lock<std::mutex> l(_row_batches_lock);
```

Comment:
You can use util/condition_variable.h.
It comes from kudu, it's use is simple than std::

Reply:
This PR is only try to solve the BE crash problem and I don't want to introduce any new potential problems.
I can't see any necessary to replace the `std::condition_variable` with `ConditionVariable`. If it is,
maybe we can replace all `std::condition_variable` in another PR.